packages/engine/lib/error/src/frame.rs

```diff
@@ -185,31 +198,62 @@ impl Drop for Frame {
 /// are stored and called directly.
 struct VTable {
     object_drop: unsafe fn(Box<FrameRepr>),
-    object_ref: unsafe fn(&FrameRepr) -> &dyn Context,
+    object_ref: unsafe fn(&FrameRepr) -> &dyn Unerased,
     object_downcast: unsafe fn(&FrameRepr, target: TypeId) -> Option<NonNull<()>>,
 }
 
-/// Wrapper around [`Message`].
+/// Wrapper around a contextual message.
 ///
-/// As [`Message`] does not necessarily implement [`Provider`], an empty implementation is provided.
-/// If a [`Provider`] is required use it directly instead of [`Message`].
-struct MessageRepr<M: Message>(M);
+/// A message does not necessarily implement [`Provider`], an empty implementation is provided.
+/// If a [`Provider`] is required attach it directly rather than attaching a message.
+struct MessageRepr<M>(M);
 
-impl<C: Message> fmt::Display for MessageRepr<C> {
-    fn fmt(&self, fmt: &mut Formatter<'_>) -> fmt::Result {
+impl<C: fmt::Display> fmt::Display for MessageRepr<C> {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
         fmt::Display::fmt(&self.0, fmt)
     }
 }
 
-impl<C: Message> fmt::Debug for MessageRepr<C> {
-    fn fmt(&self, fmt: &mut Formatter<'_>) -> fmt::Result {
+impl<C: fmt::Debug> fmt::Debug for MessageRepr<C> {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
         fmt::Debug::fmt(&self.0, fmt)
     }
 }
 
-impl<C: Message> Provider for MessageRepr<C> {
+impl<C: Context> Context for MessageRepr<C> {}
+
+#[cfg(nightly)]
+impl<C: fmt::Display + fmt::Debug + Send + Sync + 'static> Provider for MessageRepr<C> {
     fn provide<'a>(&'a self, _: &mut Demand<'a>) {
-        // Empty definition as `Message` does not convey provider information
+        // Empty definition as a contextual message does not convey provider information
+    }
+}
+
+/// Wrapper around [`Context`].
+///
+/// As [`Context`] does not necessarily implement [`Provider`], an empty implementation is provided.
+/// If a [`Provider`] is required use it directly instead of [`Context`].
+#[repr(transparent)]
+struct ContextRepr<C>(C);
```

Comment:
Required as `Context` does not implement `Provider`, but `Unerased` requires it on nightly